root.go

```diff
@@ -190,7 +196,7 @@ func (kr *Root) FlushMemFree(ctx context.Context) error {
 // the top), document it and maybe make it an anonymous variable (if
 // that's possible).
 func (kr *Root) updateChildEntry(c child) error {
-	err := kr.GetDirectory().dagService.Add(context.TODO(), c.Node)
+	err := kr.GetDirectory().dagService.Add(kr.ctx, c.Node)
```

Comment:
It seems like the context here should be passed into this function as an argument, and used to cancel _this call_ should it take too long or need to be canceled.  I guess this is to cause this `dagService.Add()` to fail if this mfs root has been closed, but adds to my confusion about calling `Close` vs canceling the context.